### PR TITLE
Fix basis order issue in molden.load (issue #1961)

### PR DESCRIPTION
**Pull request by sunqm**

### CODE REVIEW

File: pyscf/gto/mole.py [hunk 1]
```diff
@@ -503,11 +509,6 @@ def converter(symb, raw_basis):
                     _basis.extend(nparray_to_list(rawb))
         else:
             _basis = nparray_to_list(raw_basis)
-
-        # Sort basis accroding to angular momentum. This is important for method
-        # decontract_basis, which assumes that basis functions with the same
-        # angular momentum are grouped together. Related to issue #1620 #1770
-        _basis = sorted([b for b in _basis if b], key=lambda b: b[0])
```
Comment (kevfocke) [inline pyscf/gto/mole.py:510]: Oh wow, I didn't find this line when I was looking for it. Thank you so much!

File: pyscf/tools/molden.py [hunk 2]
```diff
@@ -239,8 +239,16 @@ def read_one_bas(lsym, nb, fac=1):
...
-    mol.basis = envs['basis'] = basis
     mol.atom = [atoms[i] for i in atom_seq]
+    uniq_atoms = {a[0] for a in mol.atom}
+
+    # To avoid the mol.build() sort the basis, disable mol.basis and set the
+    # internal data _basis directly. It is a workaround to solve issue #1961.
+    # Mole.decontract_basis function should be rewritten to support
+    # discontinuous bases that have the same angular momentum.
+    mol.basis = {}
+    _basis = gto.mole._parse_default_basis(basis, uniq_atoms)
```
Comment (kevfocke) [inline pyscf/tools/molden.py:250]: I guess this somewhat limits what one can do with objects retrieved from molden files, depending on how many functions depend on the sorted basis functions and how common these non standard sortings are.

File: pyscf/tools/test/test_molden.py
Comment (kevfocke) [inline pyscf/tools/test/test_molden.py:1]: The test seems sensible and simple, although it also fails, when both the basis and the mo_coeffs would hypothetically be sorted the same way successfully. So if that is ever implemented, one has to keep in mind that the tested behavior is not necessarily the perfect behavior.

**Review by kevfocke — APPROVED** [review]
Thank you so much! This is a simple fix without much additional code. 
I tested it for my use case and it works as expected.

File: pyscf/tools/molden.py (hunk 2 above)
Comment (sunqm) [reply pyscf/tools/molden.py:250]: The case of un-sorted basis was considered here. Calling mol.build() will enforce a sort in the function. However, assigning the ._basis attribute of the Mole object directly will skip the sorting. Probably a better solution is to use an input parameter in Mole object to control whether to sort the basis. Introducing the new parameters may require other updates in Mole class. To solve the issue 1961, we can use this temporary solution for the moment

**Review by fishjojo — APPROVED** [review]

**Pull request merged**